src/Microsoft.Identity.Web.DownstreamApi/DownstreamApi.cs

```diff
@@ -336,7 +356,19 @@ internal partial class DownstreamApi : IDownstreamApi
             // which is done by not specifying any scopes
             if (effectiveOptions.Scopes != null && effectiveOptions.Scopes.Any())
             {
-                string authorizationHeader = appToken ?
+                string? authorizationHeader;
+                if (_authorizationHeaderProviderExtension != null)
```

Comment:
due to dependency injection, this will not be null, so no need for the check.

Reply:
Good point! Moved away from the design of introducing a new interface so this change is so longer relevant.